### ISSUE TITLE
Consider Z-axis in a shadow generation

### ISSUE THREAD

**Issue by orangy** [BirdiePeep/UnityLight2d#3]
Current implementation (and Unity's own URP 2D system) lacks an ability to cast a shadow from one object to another object with the artificial Z-axis in mind. E.g. if I have a space ship flying over a surface, I'd like to have a shadow from the ship on the surface. Or if I have a hero in a side scroller, a shadow on the wall would make it look more deep. 

I understand, than 2.5D setup could be an easier approach to achieve this, but pure 2D setup with a shadow system capable of doing such things would be nice. 

I don't really know if it is technically possible, what are limitations, how to do the "sun" and point pseudo-3D lights, so it's more a topic to discuss than an issue.

**Comment by BirdiePeep** [comment]
Yeah I understand the need for a type of shadow that isn't always blocking the light in the 2D plane.  In my own situation for a top down game I want a few lights that cast downward.

For basic situations you can fake this with a sprite representing the shadow.  Just kind of a black overlay or something with a negative blend mode.

Though it would require another g-buffer it might be possible to have a secondary texture that was in screen space, not 1D polar space.  Shadow geometry drawn into this texture could then act as a global mask for light sources.  I'm not sure how useful that might be, it might require some sort of layer/sorting system as you might not want these masks affecting every light.
